rct229/rulesets/ashrae9012019/ruleset_functions/get_BPF_building_area_types_and_zones.py

```diff
@@ -0,0 +1,142 @@
+    LIGHTING_BUILDING_AREA.HOTEL_MOTEL: "HOTEL_MOTEL",
+    LIGHTING_BUILDING_AREA.LIBRARY: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.MANUFACTURING_FACILITY: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.MULTIFAMILY: "MULTIFAMILY",
+    LIGHTING_BUILDING_AREA.MUSEUM: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.OFFICE: "OFFICE",
+    LIGHTING_BUILDING_AREA.PARKING_GARAGE: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.PENITENTIARY: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.PERFORMING_ARTS_THEATER: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.POLICE_STATION: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.POST_OFFICE: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.RELIGIOUS_FACILITY: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.RETAIL: "RETAIL",
+    LIGHTING_BUILDING_AREA.SCHOOL_UNIVERSITY: "SCHOOL",
+    LIGHTING_BUILDING_AREA.SPORTS_ARENA: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.TOWN_HALL: "OFFICE",
+    LIGHTING_BUILDING_AREA.TRANSPORTATION: "ALL_OTHER",
+    LIGHTING_BUILDING_AREA.WAREHOUSE: "WAREHOUSE",
+    LIGHTING_BUILDING_AREA.WORKSHOP: "WAREHOUSE",
+    LIGHTING_BUILDING_AREA.NONE: "NONE",
+}
+
+
+class BuildingAreaTypeWithTotalAreaAndZone(TypedDict):
+    zone_id: list[str]
+    area: Quantity
+    classification_source: str
+
+
+def get_BPF_building_area_types_and_zones(
+    rmd: dict,
+) -> dict[str, BuildingAreaTypeWithTotalAreaAndZone]:
+    """
+    Get a dictionary of the zone_ids associated with each BPF building area type associated with U_RMD, P_RMD, or B_RMD. Also returns the total floor area of each building area type and the source of the information (BUILDING_AREA_LIGHTING or SPACE_LIGHTING)
+
+    Parameters
+    ----------
+    rmd: str, The ruleset model descriptions
+
+    Returns
+    -------
+    building_area_types_with_total_area_and_zones_dict: A dict that saves all the BPF building area types and includes a list of all the zone ids associated with area type as well as
+                                                       the total area of each building area type: {MULTIFAMILY: {"zone_id": ["zone_2", "zone_3", "zone_4"], "area": 34567, "classification_source": "BUILDING_SEGMENT_LIGHTING"}, HEALTHCARE_HOSPITAL: {"zone_id": ["zone_1","zone_5","zone_6"], "area": 20381, "classification_source": "SPACE_LIGHTING"}
+    """
+
+    building_area_types_with_total_area_and_zones_dict = {}
+    for building_segment in find_all("$.buildings[*].building_segments[*]", rmd):
+        if (
+            building_segment.get("lighting_building_area_type") is not None
+            and building_segment["lighting_building_area_type"]
+            != LIGHTING_BUILDING_AREA.NONE
+        ):
+            building_segment_BPF_BAT = LIGHTING_BUILDING_AREA_LOOKUP[
+                building_segment["lighting_building_area_type"]
+            ]
+            classification_source = "BUILDING_SEGMENT_LIGHTING"
+        else:
+            building_segment_space_types_areas_dict = {}
+            building_segment_BPF_BAT = {}
```

Comment:
A building segment can only have 1 BPF BAT assigned to it so I don't see why it is a dict in the space lighting classification method.